### ISSUE TITLE
Feature request: implementation of exclude filter

### ISSUE THREAD

**Issue by 4pathfinder** [osmandapp/OsmAnd#15315]
# 🚀 feature request

### Description

An extension of the current filter with an option of excluding certain tags from the search would be really helpful. This can best be explained using an example: I recently wanted to search for nearby vending machines. Some of the ones that I know didn't show up as they were tagged as vending=pizza, a filter not being provided by OsmAnd. To find them, I had to reset all filters and show all vending machines on the map, most of them being cigarette vending machines. It would be helpful if certain tags could explicitly be excluded, such as vending_type=cigarettes, bicycle_tubes, etc. to effectively reduce the number of POIs shown on the map and to more easily find the POIs you're interested in and that are not available as a filter in OsmAnd.

### Describe the solution you'd like

Integration of a filter to show all POIs except the selected categories. Example: Checking Vending Type = cigarettes and applying the exclude filter resulting in a map view showing all vending types except for cigarettes.

**Comment by pebogufi** [comment]
Example:
for vending machines are about 35 "subtypes", which can be selected separately.

**Comment by 4pathfinder** [comment]
Hi pebogufi

Yes I am aware of that. But just like mentioned in the example, vending_type=pizza is not contained in the filters provided by OsmAnd, which is why I got the idea of "reverse searching" for them by excluding as many negatives (= non pizza) as possible and check the remaining ones.

**Comment by vshcherb** [comment]
Yes we have that request when you would like to find something but "1 type" especially when list of subtypes is not complete than it becomes a problem cause you need to have "Other" as well. Excluding filter could be one of solution but it could make UI more complicated
